src/utilities/observables/__tests__/asyncMap.ts

```diff
@@ -141,4 +141,122 @@ describe("asyncMap", () => {
       }),
     });
   });
+
+  test.each([
+    ["sync", (n: number) => n * 2],
+    ["async", async (n: number) => n * 2],
+  ])("[%s] mapFn maps over values", async (_, mapFn) => {
+    const observable = new Observable<number>((observer) => {
+      observer.next(1);
+      observer.next(2);
+      setTimeout(() => {
+        observer.next(3);
+        setTimeout(() => {
+          observer.next(4);
+          observer.complete();
+        }, 10);
+      }, 10);
+    });
+    const mapped = asyncMap(observable, mapFn);
+    const taker = new ObservableTaker(mapped);
+    await expect(taker.takeNext()).resolves.toBe(2);
+    await expect(taker.takeNext()).resolves.toBe(4);
+    await expect(taker.takeNext()).resolves.toBe(6);
+    await expect(taker.takeNext()).resolves.toBe(8);
+    await taker.takeComplete();
+  });
+
+  test.each([["sync"], ["async"]])(
+    "[%s] mapFn can convert next to error",
```

Comment:
```suggestion
    "[%s] notifies the observer with an error when an error is thrown inside the mapFn",
```

Not quite sure the right terminology to use here ("sends error notification" maybe?), but the behavior as I understand it isn't so much about "converting next to error", but rather just that when a map function throws an error, the subscriber is notified via `error` instead of `next` (no "conversion" happening here). Could we update the test description to resemble that behavior a bit more? Feel free to change my suggestion here to match the right observable terminology, but hopefully what I'm getting at makes sense 🙂

Reply:
In my mindset, the `mapFn` actively has the role to "convert" the incoming value into an error, making sure that it takes the `error` route from there on.

To me it feels like `mapFn` and `catchFn` do some kind of "implicit conversion" here - `mapFn` could convert a normal value with `{ data: null, errors: [ ... ] }` into a real error down the line, while `catchFn` could convert a network error into a normally emitted value like `{ data: null, networkError: [] }`

Does that make sense to you?

Reply:
I think of it sort of like how promise chaining works. If an error is thrown inside a `.then()`, you're not really doing a conversion from a resolved promise to a rejected promise, but rather its the result of natural error handling within promise chaining. You're essentially just handling errors inside `.then`s by creating new rejected promises.

Similar here, `map` returns a new observable that emits values returned by the `mapFn` and errors just emit an error when encountered.

I'll leave it up to you to decide what you like better here, but I won't be upset at all if you decided to keep this. Just figured this might be an opportunity for a bit more clarity if possible, but feel free to make a judgement call on what you want here!